### ISSUE TITLE
bug: Can't archive freshly unarchived thread

### ISSUE THREAD

**Issue by phorcys420** [nyxx-discord/nyxx#427]
**Describe the bug**
I'm running into ``Bad state: Stream has already been listened to.`` when trying to archive a freshly-unarchived thread.
I belive this might be a race condition of some sort (where the channel has not yet finished being updated or whatnot).

The reason it's freshly unarchived is because I'm running a command as the first message in an archived thread.
so basically, I'm running ``/archive``, which un-archives the thread because that sent a message, and the command tries to archive it back but I feel like it's going too fast. running the command again afterwards is fine.

**To Reproduce**
```dart
final slashCommand = SlashCommandBuilder(
    "archive",
    "e",
    canBeUsedInDm: false,
  )..registerHandler((p0) async {
      final thread = await p0.interaction.channel.download() as IThreadChannel;

      ThreadBuilder threadBuilder = ThreadBuilder(name);
      threadBuilder.archived = true;
      threadBuilder.locked = false;

      thread.edit(threadBuilder);
    })

interactions.registerSlashCommand(slashCommand);
```

**Expected behavior**
The thread should archive and my command should not error.

**Desktop (please complete the following information):**
 - OS: Linux
 - Dart version: 2.18.0
 - Nyxx version: 4.4.0

**Additional context**
This can be worked around by editing with ``archive = true``, then with ``archive = false``, like so:
```dart
ThreadBuilder threadBuilder = ThreadBuilder(name);
threadBuilder.archived = false;
threadBuilder.locked = false;

thread.edit(threadBuilder);

threadBuilder.archived = true;

thread.edit(threadBuilder);
```

I don't know if this belongs in nyxx_interactions but I believe it might have to do with nyxx itself.

**Comment by abitofevrything** [comment]
Hey there, could you attach a stack trace of where this error occurs? I'm struggling to locate where the error is happening.

**Comment by phorcys420** [comment]
Sorry, here is the stack trace.
```
Error: Bad state: Stream has already been listened to.
Stack trace:
#0      _StreamController._subscribe (dart:async/stream_controller.dart:676:7)
#1      _ControllerStream._createSubscription (dart:async/stream_controller.dart:827:19)
#2      _StreamImpl.listen (dart:async/stream_impl.dart:471:9)
#3      new _ForwardingStreamSubscription (dart:async/stream_pipe.dart:114:10)
#4      _ForwardingStream._createSubscription (dart:async/stream_pipe.dart:86:16)
#5      _ForwardingStream.listen (dart:async/stream_pipe.dart:81:12)
#6      _HttpIncoming.listen (dart:_http/http_impl.dart:444:8)
#7      _HttpClientResponse.listen (dart:_http/http_impl.dart:703:19)
#8      new _ForwardingStreamSubscription (dart:async/stream_pipe.dart:114:10)
#9      _ForwardingStream._createSubscription (dart:async/stream_pipe.dart:86:16)
#10     _ForwardingStream.listen (dart:async/stream_pipe.dart:81:12)
#11     StreamView.listen (dart:async/stream.dart:2201:20)
#12     new _SinkTransformerStreamSubscription (dart:async/stream_transformers.dart:49:16)
#13     _BoundSinkStream.listen (dart:async/stream_transformers.dart:171:9)
#14     Stream.fold (dart:async/stream.dart:1091:14)
#15     Encoding.decodeStream (dart:convert/encoding.dart:24:10)
#16     ByteStream.bytesToString (package:http/src/byte_stream.dart:33:16)
#17     HttpHandler._handle (package:nyxx/src/internal/http/http_handler.dart:125:61)
<asynchronous suspension>
#18     HttpEndpoints.executeSafe (package:nyxx/src/internal/http_endpoints.dart:453:22)
<asynchronous suspension>
#19     HttpEndpoints.editThreadChannel (package:nyxx/src/internal/http_endpoints.dart:1819:9)
<asynchronous suspension>
#20     handleResolve (package:codercord/discord/commands/resolve.dart:15:5)
<asynchronous suspension>
#21     getCommand.<anonymous closure> (package:codercord/discord/commands/resolve.dart:101:7)
<asynchronous suspension>
#22     Interactions.sync.<anonymous closure> (package:nyxx_interactions/src/interactions.dart:262:11)
<asynchronous suspension>
```

**Comment by phorcys420** [comment]
Different version, different issue. Here's with 4.4.1
```
Error: Unknown interaction (10062) POST https://discord.com/api/v10/interactions/1066999320957693982/[...]/callback
Stack trace:
#0      HttpEndpoints.executeSafe (package:nyxx/src/internal/http_endpoints.dart:456:48)
<asynchronous suspension>
#1      InteractionsEndpoints.executeSafe (package:nyxx_interactions/src/internal/interaction_endpoints.dart:125:22)
<asynchronous suspension>
#2      InteractionEventWithAcknowledge.respond (package:nyxx_interactions/src/events/interaction_event.dart:229:7)
<asynchronous suspension>
#3      handleResolve (package:codercord/discord/commands/resolve.dart:50:11)
<asynchronous suspension>
#4      getCommand.<anonymous closure> (package:codercord/discord/commands/resolve.dart:101:7)
<asynchronous suspension>
#5      Interactions.sync.<anonymous closure> (package:nyxx_interactions/src/interactions.dart:262:11)
<asynchronous suspension>
```

**Comment by abitofevrything** [comment]
This seems to be a separate issue. You can close this one and reopen a new one (probably on the nyxx_interactions repo, looking at the stack trace), or ask for help on the Discord server.

**Comment by l7ssha** [comment]
Core of the issue fixed in 4.4.1 . As @abitofevrything you can open separate issue in nyxx_interactions if the problem remains :)
